### PR TITLE
Add Author DTO and Implement POST Endpoint

### PR DESCRIPTION
**Pull request by Qubicon**

### CODE REVIEW

File: src/main/java/ro/unibuc/hello/controller/AuthorController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Autowired
+    private AuthorService authorService;
+
+    @PostMapping("/authors")
+    @ResponseBody
+    public ResponseEntity<AuthorEntity> createAuthor(@RequestBody AuthorCreationRequestDto authorCreationRequestDto) {
+        String name = authorCreationRequestDto.getName();
+        String nationality = authorCreationRequestDto.getNationality();
+        var newAuthor = authorService.saveAuthor(name, nationality);
```
Comment (irinaenescu2002) [inline src/main/java/ro/unibuc/hello/controller/AuthorController.java]: Putem sa folosim direct ```var newAuthor = authorService.saveAuthor(authorCreationRequestDto.getName(), authorCreationRequestDto.getNationality());```, nu e necesar sa le salvam in variabile diferite.

File: src/main/java/ro/unibuc/hello/service/AuthorService.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import ro.unibuc.hello.data.AuthorRepository;
+
+@Service
+@RequiredArgsConstructor
+public class AuthorService {
+
+    @Autowired
+    private AuthorRepository authorRepository;
+
+    public AuthorEntity saveAuthor(String name, String nationality) {
```
Comment (irinaenescu2002) [inline src/main/java/ro/unibuc/hello/service/AuthorService.java:20]: Cred ca este o idee buna sa adaugam log-uri inca de pe acum, asa ca: 
1. Adauga adnotarea ```@Slf4j```
2. Adauga un log parametrizat de debug 
```log.debug("Creating a new author '{}' with nationality '{}'", name, nationality);```

**Review by irinaenescu2002 — CHANGES_REQUESTED** [review]
Doua modificari mici

**Comment by Qubicon** [comment]
```
Reduce all commits to 1 on your feature branch before merging the PR (using rebase and squash).
```

So use **_Squash and merge_**.

**Review by irinaenescu2002 — APPROVED** [review]
Gj

**Pull request merged**